src/process.rs

```diff
@@ -134,11 +137,147 @@
 ///            Expression::Paren(Box::new(Expression::Paren(Box::new(Expression::Letter('z'))))));
 /// # }
 /// ```
+///
+/// ### Sentence
+///
+/// To showcase the use of multiple matchers, we'll use a setence grammar:
+///
+/// ```no_run
+/// # #[macro_use] extern crate pest;
+/// # use pest::prelude::*;
+/// # fn main() {
+/// # impl_rdp! {
+/// # grammar! {
+/// sentence = { word ~ ([" "] ~ word)* }
+/// word     = { letter* }
+/// letter   = { ['a'..'z'] }
+/// # }
+/// # }
+/// # unreachable!();
+/// # }
+/// ```
+///
+/// A very simple AST that works in this case:
+///
+/// ```no_run
+/// # use std::collections::LinkedList;
+/// #[derive(Debug, PartialEq)]
+/// pub enum Node {
+///     Sentence(LinkedList<Node>), // we're using LinkedList because they're more efficient when
+///     Word(LinkedList<Node>),     // using tail recursion
+///     Letter(char)
+/// }
+/// ```
+///
+/// To build the `Token` processor, let's use a bottom up aproach when writing the matchers. Let's
+/// start by building a `word` matcher. We'll call it `_word` in order not to clash with the name
+/// of the rule that also gets defined as a method on the `Parser`.
+///
+/// ```ignore
+/// _word(&self) -> LinkedList<Node> { // return LinkedList<Node> to buil Word with
+///     (&head: letter, @tail) => { // usual tail recursion; &head is captured,
+///         let mut tail = tail;    // tail is recursive
+///         tail.push_front(Node::Letter(head.chars().next().unwrap()));
+///
+///         tail
+///     },
+///     () => {                // if rule above doesn't match, there are no more letters to
+///         LinkedList::new()  // process; return empty list
+///     }
+/// }
+/// ```
+///
+/// Processing a `sentence` is similar, only this time `head` will be a `_word` call.
+///
+/// ```ignore
+/// _sentence(&self) -> LinkedList<Node> {
+///     (_: word, @<_word>head, @tail) => { // match word Token then call _word
```

Comment:
I know. Any better ideas for it? I was thinking of changing the `@` rule to `head = _word()` and making the function that you want to call explicit. What do you think?